pkg/manager/member/utils.go

```diff
@@ -262,3 +263,36 @@ func MapContainers(podSpec *corev1.PodSpec) map[string]corev1.Container {
 	}
 	return m
 }
+
+// updateStatefulSet is a template function to update the statefulset of components
+func updateStatefulSet(setCtl controller.StatefulSetControlInterface, tc *v1alpha1.TidbCluster, newSet, oldSet *apps.StatefulSet) error {
+	isOrphan := metav1.GetControllerOf(oldSet) == nil
+	if !statefulSetEqual(*newSet, *oldSet) || isOrphan {
+		set := *oldSet
+		// Retain the deprecated last applied pod template annotation for backward compatibility
+		var podConfig string
+		var hasPodConfig bool
+		if oldSet.Spec.Template.Annotations != nil {
+			podConfig, hasPodConfig = oldSet.Spec.Template.Annotations[LastAppliedConfigAnnotation]
+		}
+		set.Spec.Template = newSet.Spec.Template
+		if hasPodConfig {
+			set.Spec.Template.Annotations[LastAppliedConfigAnnotation] = podConfig
+		}
```

Comment:
`SetStatefulSetLastAppliedConfigAnnotation` is used to set last-applied-config annotation(s), move the `Spec.Template.Annotations[LastAppliedConfigAnnotation]` backward compatility logic into it?

Reply:
It will change the interface of this method (which needs the old annotation)